server/tests/src/test/java/org/infinispan/server/xsite/XSiteExample.java

```diff
@@ -0,0 +1,182 @@
+   private static Map<List<Boolean>, Integer> expectedValueTable = new HashMap<>();
+
+   @ClassRule
+   public static final InfinispanXSiteServerRule SERVERS = XSiteIT.SERVERS;
+
+   @Rule
+   public InfinispanXSiteServerTestMethodRule SERVER_TEST = new InfinispanXSiteServerTestMethodRule(SERVERS);
+
+   private boolean offHeap;
+   private boolean maxIdle;
+   private boolean store;
+   private boolean eviction;
+
+   public XSiteExample(boolean offHeap, boolean maxIdle, boolean store, boolean eviction) {
+      this.offHeap = offHeap;
+      this.maxIdle = maxIdle;
+      this.store = store;
+      this.eviction = eviction;
+   }
+
+   @Parameterized.Parameters(name = "offHeap={0}, maxIdle={1}, store={2}, eviction={3}")
+   public static Collection<Boolean[]> data() {
+      List<Boolean[]> params = new ArrayList<>();
+      boolean[] configs = {true, false};
+      for (boolean offHeap : configs) {
+         for (boolean maxIdle : configs) {
+            for (boolean store : configs) {
+               for (boolean eviction : configs) {
+                  if (offHeap && eviction) {
+                     continue;
+                  }
+                  params.add(new Boolean[]{offHeap, maxIdle, store, eviction});
+
+                  Integer expectedValue = 1000;
+                  if (eviction) {
+                     expectedValue = 500;
+                  }
+                  if (offHeap) {
+                     expectedValue = 10;
+                  }
+                  // custom
+                  if (offHeap && store) {
+                     expectedValue = 1000;
+                  }
+                  if (eviction && store) {
+                     expectedValue = 1000;
+                  }
+                  // last
+                  if (maxIdle) {
+                     expectedValue = 0;
+                  }
+                  expectedValueTable.put(Arrays.asList(offHeap, maxIdle, store, eviction), expectedValue);
+               }
+            }
+         }
+      }
+      return params;
+   }
+
+   @Test
+   public void testHotRodOperations() {
+
+      RemoteCache<String, String> lonCache = createCacheConfig(offHeap, maxIdle, store, eviction, "LON", "NYC");
+      RemoteCache<String, String> nycCache = createCacheConfig(offHeap, maxIdle, store, eviction, "NYC", "LON");
+
+      List<Boolean> key = Arrays.asList(offHeap, maxIdle, store, eviction);
+      Integer expectedValue = expectedValueTable.get(key);
+      insertAndVerifyEntries(lonCache, nycCache, expectedValue);
+   }
+
+   private void insertAndVerifyEntries(RemoteCache<String, String> lonCache, RemoteCache<String, String> nycCache, Integer expectedValue) {
+      for (int i = 0; i < 1_000; i++) {
+         lonCache.put("key-" + i, "value-" + i);
+      }
+
+      // there is a reaper that runs
+      if (maxIdle) {
+         sleep(70_000);
+      } else {
+         sleep(5_000); // sleep isn't a bad thing, right ?
```

Comment:
I added this because I was thinking that it could have a `reaper ` only when store was enabled
`[offHeap=false, maxIdle=false, store=true, eviction=false]`
It is not related and we should remove